producer_consumer/producer_consumer.c

```diff
@@ -653,7 +653,8 @@ static void *consumer(void *arg)
 
 int idx_arr_size = INDEX_ARRAY_SIZE;
 int data_arr_size =  DATA_ARRAY_SIZE;
-int cpu_producer = -1, cpu_consumer = -1;
+int *cpu_producer, *cpu_consumer;
```

Comment:
So `cpu_producer` and `cpu_consumer` will be NULL initially. which can be taken to mean use all the CPUs :)

Reply:
yeah this works for the empty cases too. With the assumption that no argument means, all cpus are being used.